File: src/test/java/kitchenpos/fixtures/OrderFixtures.java
package kitchenpos.fixtures;

import kitchenpos.domain.Menu;
import kitchenpos.domain.Order;
import kitchenpos.domain.OrderLineItem;
import kitchenpos.domain.OrderStatus;
import kitchenpos.domain.OrderTable;
import kitchenpos.domain.OrderType;

import java.time.LocalDateTime;
import java.util.Arrays;
import java.util.Random;
import java.util.UUID;

public class OrderFixtures {

    public static Order order(final OrderStatus status, final OrderTable orderTable, final Menu menu) {
        final Order order = new Order();
        order.setId(UUID.randomUUID());
        order.setType(OrderType.EAT_IN);
        order.setStatus(status);
        order.setOrderDateTime(LocalDateTime.of(2025, 2, 1, 12, 0));
        order.setOrderLineItems(Arrays.asList(orderLineItem(menu)));
        order.setOrderTable(orderTable);
        return order;
    }

    public static OrderLineItem orderLineItem(final Menu menu) {
        final OrderLineItem orderLineItem = new OrderLineItem();
        orderLineItem.setSeq(new Random().nextLong());
        orderLineItem.setMenu(menu);
        return orderLineItem;
    }
}

File: src/testFixtures/kotlin/domain/OrderFixtures.kt
package domain

import domain.OrderLineItemFixtures.makeOrderLineItemOne
import kitchenpos.domain.Order
import kitchenpos.domain.OrderStatus
import kitchenpos.domain.OrderType
import java.time.LocalDateTime
import java.util.*

object OrderFixtures {
    fun makeOrderOne(): Order {
        return Order().apply {
            this.id = UUID.fromString("69d78f38-3bff-457c-bb72-26319c985fd8")
            this.type = OrderType.DELIVERY
            this.status = OrderStatus.SERVED
            this.deliveryAddress = "서울시 강남구"
            this.orderLineItems = listOf(makeOrderLineItemOne())
            this.orderDateTime = LocalDateTime.now()
        }
    }

    fun makeOrderTwo(): Order {
        return Order().apply {
            this.id = UUID.fromString("4519efc6-91af-446c-b54f-f1f5e4d11201")
            this.type = OrderType.EAT_IN
            this.status = OrderStatus.COMPLETED
            this.orderTable = OrderTableFixtures.makeOrderTableOne()
            this.orderLineItems = listOf(makeOrderLineItemOne())
            this.orderDateTime = LocalDateTime.now()
        }
    }
}
